src/views/projectExplorer/variables.ts

```diff
@@ -2,15 +2,44 @@
  * (c) Copyright IBM Corp. 2023
  */
 
-import { ThemeIcon, TreeItem, TreeItemCollapsibleState, Uri, WorkspaceFolder } from "vscode";
+import { ThemeIcon, TreeItemCollapsibleState, Uri, WorkspaceFolder } from "vscode";
+import { ProjectExplorerTreeItem } from "./projectExplorerTreeItem";
+import { ProjectManager } from "../../projectManager";
+import ErrorItem from "./errorItem";
+import Variable from "./variable";
+import { ContextValue } from "../../projectExplorerApi";
+
+export default class Variables extends ProjectExplorerTreeItem {
+  static contextValue = ContextValue.variables;
 
-export default class Variables extends TreeItem {
-  static contextValue = `variables`;
   constructor(public workspaceFolder: WorkspaceFolder, unresolvedVariableCount: number) {
     super(`Variables`, TreeItemCollapsibleState.Collapsed);
 
     this.resourceUri = Uri.parse(`variables:${unresolvedVariableCount}`, true);
     this.contextValue = Variables.contextValue;
     this.iconPath = new ThemeIcon(`symbol-variable`);
   }
+
+  async getChildren(): Promise<ProjectExplorerTreeItem[]> {
+    let items: ProjectExplorerTreeItem[] = [];
+
+    const iProject = ProjectManager.get(this.workspaceFolder);
+    const possibleVariables = iProject?.getVariables();
+    const actualValues = await iProject?.getEnv();
+
+    if (possibleVariables && actualValues) {
+      items.push(...possibleVariables?.map(
+        varName => {
+          return new Variable(this.workspaceFolder, varName, actualValues[varName]);
+        }
+      ));
+
```

Comment:
So if there is a variable with no value in the env, how do we represent that variable

Reply:
If there is a variable with no value, the description will indicate this. There is also a red decoration on the tree item to make it clear.

![image](https://user-images.githubusercontent.com/32170854/235489707-50499eec-ef13-41f7-a9b1-d88e8c30bf47.png)